### PR TITLE
[OBSDEF-5242] added crds

### PR DESCRIPTION
**Pull request by alexander-rassanov**
## Purpose
[OBSDEF-5242](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-5242)
Added CRDS

## PR checklist
- [ ] make test passed
- [ ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing
_Paste the output of your helm install/vSphere7 deployment testing here_

### CODE REVIEW

**Review by hoklet — APPROVED** [review]

File: objectscale-manager/crds/storagepolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: storagepolicies.ecs.dellemc.com
```
Comment (eddiepavkovic) [inline objectscale-manager/crds/storagepolicy.yaml:5]: we need to rename this,  it conflicts with storagepolicies on vsphere.

```
─ $ ▶  kubectl get storagepolicies
NAME                                    AGE
dellemc-objectscale-highly-available    13h
dellemc-objectscale-vsan-direct-thick   13h
dellemc-objectscale-vsan-sna-thick      13h

```

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]

File: objectscale-manager/crds/storagepolicy.yaml (hunk 1 above)
Comment (alexander-rassanov) [reply objectscale-manager/crds/storagepolicy.yaml:5]: fixed

Comment (eddiepavkovic) [reply objectscale-manager/crds/storagepolicy.yaml:5]: but this name is still storagepolicies...  which kinda conflicts with vsphere storagepolicies...  
```suggestion
  name: storagepolicies.ecs.dellemc.com
```

File: objectscale-manager/crds/storagepolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    app.kubernetes.io/component: objectscale-operator
+    app.kubernetes.io/part-of: objectscale-manager
+    app.kubernetes.io/managed-by: nautilus
+spec:
+  group: ecs.dellemc.com
+  names:
+    kind: StoragePolicy
+    listKind: StoragePolicyList
+    plural: ecs-storagepolicies
+    singular: ecs-storagepolicy
```
Comment (eddiepavkovic) [inline objectscale-manager/crds/storagepolicy.yaml:18]: jason wants us to not use `ecs` in the charts.  can we use `objectscale-storagepolicies`
```suggestion
    singular: ecs-storagepolicy
```

File: objectscale-manager/crds/storagepolicy.yaml (hunk 1 above)
Comment (alexander-rassanov) [reply objectscale-manager/crds/storagepolicy.yaml:5]: However we have group ECS

File: objectscale-manager/crds/storagepolicy.yaml (hunk 2 above)
Comment (alexander-rassanov) [reply objectscale-manager/crds/storagepolicy.yaml:18]: fixed

File: objectscale-manager/crds/storagepolicy.yaml (hunk 1 above)
Comment (alexander-rassanov) [reply objectscale-manager/crds/storagepolicy.yaml:5]: is it ok for you?

**Review by fedor-chemashkin — APPROVED** [review]

**Review by eddiepavkovic — DISMISSED** [review]
Alex i brought this up during our standup and they agreed we shouldn't be re-using a name already taken by VMware.  can we prefix this with something more unique or come up with a different name.

**Comment by alexander-rassanov** [comment]
> Alex i brought this up during our standup and they agreed we shouldn't be re-using a name already taken by VMware. can we prefix this with something more unique or come up with a different name.

I changed the name and added prefix objectstore. The common name of the CRD is under dell.ecs group

**Pull request merged**